src/pages/ProductDetail/ProductDetail.js

```diff
@@ -1,7 +1,31 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import Rightside from './Rightside';
+import Leftside from './Leftside';
+import './ProductDetail.scss';
 
 const ProductDetail = () => {
-  return <div>ProductDetail</div>;
+  const { productId } = useParams();
+
+  const [productData, setProductData] = useState({});
+  useEffect(() => {
+    fetch(`http://10.58.52.122:3000/products/${productId}`, {
+      method: 'GET',
+    })
+      .then(response => response.json())
+      .then(data => setProductData(data.data[0]));
+  }, []);
+  return (
+    <>
+      <div className="container">
+        <div className="detail-main">
+          <Leftside productData={productData} />
+
+          <Rightside productData={productData} />
```

Comment:
productData를 서버에서 받아오기 전에는 빈 객체를 보내기 때문에 Leftside, Rightside 각 컴포넌트에서 데이터가 존재할 때의 조건부 처리를 해줘야 할 것 같은데, 아예 보내기 전에 다음과 같이 처리해버릴 수도 있습니다.
```suggestion
  const [productData, setProductData] = useState({});
  useEffect(() => {
    fetch(`http://10.58.52.122:3000/products/${productId}`, {
      method: 'GET',
    })
      .then(response => response.json())
      .then(data => setProductData(data.data[0]));
  }, []);
  
  if (Object.keys(productData).length === 0) return null;
  
  return (
    <>
      <div className="container">
        <div className="detail-main">
          <Leftside productData={productData} />

          <Rightside productData={productData} />
```
productData가 key값을 가지지 않은, 즉 빈 객체일 때, 다시 말하면 데이터가 아직 들어오지 않았을 때에는 컴포넌트를 렌더하지 않고 데이터를 받아온 다음에 렌더하도록 하면 예외처리를 따로 할 필요가 없게 됩니다.